### PR TITLE
Add stop function to Client

### PR DESCRIPTION
**Pull request by eduardo-cunha-bose**
Added stop function to Client which closes the UDP socket if it is
opened. This way the Client stops receiving data from the network
when needed.

### CODE REVIEW

File: lib/mdns/client.ex [hunk 1]
```diff
@@ -70,6 +74,14 @@ defmodule Mdns.Client do
     {:reply, state.devices, state}
   end
 
+  def handle_call(:stop, _from, state) do
```
Comment (entone) [inline lib/mdns/client.ex]: can we do the pattern matching in the function head for the udp variable?

something about setting `udp` to nil outside of the if logic bugs me.

something like this

```elixir
def handle_call(:stop, _from, %State{udp: nil} = state), do: {:reply, :ok, state}

def handle_call(:stop, _from, %State{udp: udp} = state) do
  :ok = :gen_udp.close(udp)
  {:reply, :ok, %State{state | udp: nil}}
end
```

**Review by entone — CHANGES_REQUESTED** [review]

File: lib/mdns/client.ex (hunk 1 above)
Comment (eduardo-cunha-bose) [reply lib/mdns/client.ex]: Sure, I just did it the way it was done in the Server module.
Do you want me to change it there also?

Comment (entone) [reply lib/mdns/client.ex]: that would be great. not sure how that snuck in there :)

Comment (eduardo-cunha-bose) [reply lib/mdns/client.ex]: Done :)

**Comment by entone** [comment]
Great, give me a little while to run the tests and put a release together.

Also, thank you for all the contributions!

**Comment by entone** [comment]
version 1.0.8 has been pushed to hex

**Pull request merged**